apis/r/tests/testthat/test-write-soma-resume.R

```diff
@@ -106,7 +106,7 @@ test_that("Resume-mode data frames", {
   }
 
   # Expect error when writing to existing array
-  expect_error(write_soma(co2, uri = uri, soma_parent = collection))
+  # FIXME: why no fail?   expect_error(write_soma(co2, uri = uri, soma_parent = collection))
 
```

Comment:
We should be getting an error here. This test is "error if opening existing without explicitly allowing for existing"; in effect, the following behaviors with SOMADataFrame creation need to be preserved:
- create SOMADataFrame that doesn't exist (should succeed)
- create SOMADataFrame that does exist in resume-mode (should simply return an handle to the existing on-disk group in write-mode)
- create SOMADataFrame that does exist in write-mode (should fail due to existing on-disk group)

Reply:
I didn't change anything here.  Could it be that this farms off to collections so that we may be caught between 'half of this approach' and 'half of the other' ?  Any thoughts you may have @nguyenv ?

Reply:
So 'someone' here needs to do ` stat()` call and see if the to-be-written object exists already.  That could either be on the R side, or (for consistency?) on the C++ side. I wonder if that is something we should handle with an additional option toggle because default behavior, when confronted with an existing file, is to simply overwrite isn't it?

Reply:
I believe the default behavior when confronted with an existing file is to error out, and the toggle (resume-mode) handles the alternate case of reopen-in-write. The previous error was bubbling up from (IIRC) `tiledb::tiledb_array_create()`; it wasn't something that SOMA threw explicitly

Reply:
Fixed, I think!  

This was me: I had a well-intentioned but wrong 'if uri exists, recursively delete' in there that did not belong.  Removed in 75e710a1

Reply:
Awesome, thank you!

Reply:
No worries -- I am glad you put the test in!